server/gc/src/main/java/org/apache/accumulo/gc/GCRun.java

```diff
@@ -462,4 +479,28 @@ public long getErrorsStat() {
   public long getCandidatesStat() {
     return candidates;
   }
+
+  @Override
+  public boolean isRootTable() {
+    return level == DataLevel.ROOT;
+  }
+
+  @Override
+  public boolean isMetadataTable() {
+    return level == DataLevel.METADATA;
+  }
+
+  @Override
+  public Set<TableId> getCandidateTableIDs() {
+    if (isRootTable()) {
+      return Collections.singleton(MetadataTable.ID);
+    } else if (isMetadataTable()) {
```

Comment:
```suggestion
    if (level == ROOT) {
       return Set.of(RootTable.ID);
    } else if(level == METADATA){
      return Set.of(MetadataTable.ID);
    } else if (level == USER) {
```

Reply:
This change does not match the intended behavior described in the javadoc for the method.

Reply:
I missed that javadoc, looking at it now it would need to be adjusted.  There are three data levels and we run GC for each level.  They are created here

https://github.com/apache/accumulo/blob/1aeffcf0dca3e9797d3bfd75dd05b821885b1dd4/server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java#L207-L209

So for each level we need to return the table IDs that would be seen in the metadata for that level.  Like the USER level stores its metadata in the metadata table, but we see user table ids in the metadata for that level.  For the METADATA level its stores its metadata in the root table and we see the metadata table id.  For the ROOT level it stores its metadata in ZK and we see the root table id in the metadata for that level.

Reply:
So, this PR is a port of https://github.com/apache/accumulo/pull/2293 to 2.1. Looking back through the comments there, this method is used solely to check that the tableIds before and after the scan are the same. Getting the candidates for removal is done in another method which I think has not changed. Looked at the other PR, you commented on it and said it looked ok.

Reply:
I have not looked at that PR in a long time.   The DataLevel enum did not exist in 1.10 and the GC was different.  In 1.10 the Accumulo GC process did not handle the the root tablet, it was treated specially in tserver code.  Changes were made to have less special cases for the root tablet and store its metadata in ZK and treat it like a normal tablet.  Part of these changes made the GC process handle the root tablet.  So I am thinking the changes I suggested are good for the new way the GC works in 2.1. I will look at this again in depth in an IDE tomorrow to verify I am on the right track.

Reply:
I looked how the references are obtained and I think the suggestion is correct.  In GcRun when the DataLevel is USER it will read references from the metadata table which contains user table ids.  When the DataLevel is METADATA it will read references from the root table which will contain metadata table id.  When the DataLevel is ROOT it will read the single tablet metadata entry for the root table from ZK (this is new after 1.x) which will contain root table id.

In 1.1 things operated a bit differently, there were two metadata tables to read from.  In 1.1 we would look at the table being read from an infer what kinds of table ids it contained.  Now there are two metadata tables and a single tablet metadata entry in ZK.  DataLevel is used  to specify to ample which type of metadata to read for these three data levels.

Unrelated to this PR, looking at the GCRun ref code I would like to clean it up to remove the conditional on DataLevel.Root and just pass DataLevel to ample.  I think that is an indication of the a problem in the internal ample API.

I made suggestion to update the javadoc.

Reply:
Implemented in a61969020f7ca916465811bac22b52874c3f02cd.